loadtest/control/simulcontroller/actions.go

```diff
@@ -2073,3 +2073,42 @@ func (c *SimulController) openPermalink(u user.User) control.UserActionResponse
 
 	return control.UserActionResponse{Info: fmt.Sprintf("clicked permalink on post %s", postID)}
 }
+
+func (c *SimulController) generateUserReport(u user.User) control.UserActionResponse {
+	// Simulate scrolling through the entire list of users
+	// (should be similar to generating the complete report and exporting it)
+
+	if isAdmin, err := u.IsSysAdmin(); err != nil || !isAdmin {
```

Comment:
@agnivade @agarciamontoro Is this the correct way to check for sys admin? I feel like given the frequency of running this function, is it likely that a user would be a sysadmin? Not sure if I'm understanding this correctly or not.

Reply:
Hmm .. this is a problem. This is actually exposed only in the system console right? Then I think we are doing this the wrong way. We need to simulate a full system console page load, and then navigate to the user reporting section.

Moreover, in the simulcontroller, all the users are normal users, not sysadmins. 

We can ignore the system console load for the time being. But to actually make this call work, we need to randomly bump some users to sysadmins. To achieve that, we need to modify this code: https://github.com/mattermost/mattermost-load-test-ng/blob/9fe059e92a159617b2787004c3e47ae6633e4569/api/agent.go#L387-L389 to simply take https://github.com/mattermost/mattermost-load-test-ng/blob/9fe059e92a159617b2787004c3e47ae6633e4569/api/agent.go#L433-L435 instead. 

And we can do this once every 50 users or so. So something like `if id %50 == 0 { // use sysadmin credentials} ` would work.

I think this should be acceptable for now. cc @streamer45 as well for thoughts.

Reply:
I agree with the above. I'd probably expect a new configuration setting under [`UsersConfiguration`](https://github.com/mattermost/mattermost-load-test-ng/blob/9fe059e92a159617b2787004c3e47ae6633e4569/loadtest/config.go#L126) to control how many admins to run, either as an absolute number or percentage of total users.

One important thing to note is that having an action that results in a no-op for most users will skew other frequencies so I'd probably queue the action into the [list](https://github.com/mattermost/mattermost-load-test-ng/blob/9fe059e92a159617b2787004c3e47ae6633e4569/loadtest/control/simulcontroller/controller.go#L18) only if the controller is an admin vs having it in there for all users and returning early if not admin.

Also, we should create a ticket for the system console load which I agree to be out of scope for now, something our team can probably take care of.

@devinbinnie Please let me know if you need help with the above. I realize some of this boilerplate should have probably been there already, you just happen to be a bit of a pioneer in this instance :p

Reply:
@agnivade @streamer45 Have a look :)